services/spar/test-integration/Util/Core.hs

```diff
@@ -750,44 +765,24 @@ getSsoidViaSelf uid = maybe (error "not found") pure =<< getSsoidViaSelf' uid
 
 getSsoidViaSelf' :: HasCallStack => UserId -> TestSpar (Maybe UserSSOId)
 getSsoidViaSelf' uid = do
-  musr <- getSelf uid
+  musr <- runSpar $ Intra.getBrigUser uid
   pure $ case userIdentity =<< musr of
             Just (SSOIdentity ssoid _ _) -> Just ssoid
             Just (FullIdentity _ _)  -> Nothing
             Just (EmailIdentity _)   -> Nothing
             Just (PhoneIdentity _)   -> Nothing
             Nothing                  -> Nothing
 
-getSelf :: HasCallStack => UserId -> TestSpar (Maybe User)
-getSelf uid = do
-  let probe :: HasCallStack => TestSpar (Maybe User)
-      probe = do
-        env <- ask
-        fmap selfToUser . call . get $
-          ( (env ^. teBrig)
-          . header "Z-User" (toByteString' uid)
-          . path "/self"
-          . expect2xx
-          )
-
-      selfToUser :: HasCallStack => ResponseLBS -> Maybe User
-      selfToUser (fmap Aeson.eitherDecode . responseBody -> Just (Right selfprof))
-        = Just $ selfUser selfprof
-      selfToUser _ = Nothing
-
-  env <- ask
-  liftIO $ retrying
-    (exponentialBackoff 50 <> limitRetries 5)
-    (\_ -> pure . isNothing)
-    (\_ -> probe `runReaderT` env)
-
 getUserIdViaRef :: HasCallStack => UserRef -> TestSpar UserId
 getUserIdViaRef uref = maybe (error "not found") pure =<< getUserIdViaRef' uref
 
 getUserIdViaRef' :: HasCallStack => UserRef -> TestSpar (Maybe UserId)
 getUserIdViaRef' uref = do
-  env <- ask
-  liftIO $ retrying
-    (exponentialBackoff 50 <> limitRetries 5)
-    (\_ -> pure . isNothing)
-    (\_ -> runClient (env ^. teCql) $ Data.getSAMLUser uref)
+  eventually (view teCql >>= \cql -> runClient cql $ Data.getSAMLUser uref) isNothing
+
+-- | FUTUREWORK: arguably this function should move to Util.Scim, but it also is related to
+-- the other lookups above into the various user tables in the various cassandras.  we should
+-- probably clean this up a little, and also pick better names for everything.
+getScimUser :: HasCallStack => UserId -> TestSpar (Maybe (ScimC.User.StoredUser SparTag))
+getScimUser uid = do
+  eventually (view teCql >>= \cql -> runClient cql $ Data.getScimUser uid) isNothing
```

Comment:
Why is `Nothing` considered good here?

Reply:
I'll improve the docs again.  This seems to be a very confusion helper.  :)

Reply:
actually, it was just wrong.  should be fixed now?